### ISSUE TITLE
Worker node not join to EKS cluster 1.17

### ISSUE THREAD

**Issue by phuthinh2202** [terraform-aws-modules/terraform-aws-eks#994]
# I have issues 
I use module with tag v12.2.0 and create eks cluster with version 1.17, when all task completed, i cannot see worker node when type command `kubectl get nodes`
but when create eks cluster with version 1.15 then worker node join success. what wrong on me?
## I'm submitting a...

* [x] bug report
* [ ] feature request
* [ ] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

```
module "eks" {
  source  = "terraform-aws-modules/eks/aws/"
  version = "12.2.0"

  cluster_name = "eks-name"
  vpc_id       = "......"
  subnets      = [...]


  cluster_version                 = "1.17" -> change 1.15, it's work.
  cluster_endpoint_private_access = "true"
  cluster_endpoint_public_access  = "true"

  map_roles = [
    ...
  ]

  worker_groups_launch_template = [
    {
      name                    = "example"
      override_instance_types = ["c4.2xlarge", "c5.2xlarge", "m5.2xlarge"]
      spot_instance_pools     = 4
      asg_max_size            = 4
      asg_desired_capacity    = 2
      asg_min_size            = 2
      kubelet_extra_args      = "--node-labels=kubernetes.io/lifecycle=spot"
      public_ip               = false
      autoscaling_enabled     = true
      root_volume_size        = 50
      pre_userdata            = "exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1"

  
    }
  ]

 
}
```

## Environment details

* Affected module version:
* OS: CentOS
* Terraform version: 0.12.24

**Comment by hameno** [comment]
Same issue here

**Comment by dpiddockcmp** [comment]
This is due to a feature/restriction implemented in kubernetes 1.16 as explained by gertjangaillet in https://github.com/terraform-aws-modules/terraform-aws-eks/issues/856#issuecomment-623155710

The `kubernetes.io` label namespace now has heavy restrictions. You need to use something else for your label. The linked issue suggested changing to `node.kubernetes.io/lifecycle=spot`

**Comment by phuthinh2202** [comment]
thanks. it's worked. Maybe my old code create cluster from version 1.14 and when i create new cluster with version 1.17 not work.
